### PR TITLE
Converts jobs and admin equipment to use outfits

### PR DESCRIPTION
**Pull request by Arrow768**
Converts the jobs and admin equipment to use outfit datums.

ToDo:
- [x] Testing
- [x] Adding some requested equipments
- [x] Fix missing custom items in char preview
- [ ] Possibly overhauling the admin equipment menu

### CODE REVIEW

**Comment by VTCobaltblood** [comment]
What does this mean, exactly?

File: code/datums/outfits/outfit.dm [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+			var/number = backpack_contents[path]
+			for(var/i in 1 to number)
+				H.equip_or_collect(new path(H), slot_in_backpack)
+
+	post_equip(H, visualsOnly)
+
+	if(!visualsOnly)
+		apply_fingerprints(H)
+
+	if(implants)
```
Comment (skull132) [inline code/datums/outfits/outfit.dm]: Shouldn't this be guarded by `visualsOnly` as well? Considering implants have no visual effect.

File: code/datums/outfits/outfit.dm [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+				I.imp_in = H
+				I.implanted = 1
+				var/obj/item/organ/external/affected = H.get_organ("head")
+				affected.implants += I
+				I.part = affected
+
+	H.update_body()
+	return 1
+
+/datum/outfit/proc/apply_fingerprints(mob/living/carbon/human/H)
```
Comment (skull132) [inline code/datums/outfits/outfit.dm]: Might it be easier to have this done in `equip_item`? It'd be one less giant if run.

File: code/game/jobs/job/civilian_chaplain.dm [hunk 3]
```diff
@@ -12,151 +12,152 @@
...
+			else
+				B.name = "The Holy Book of [new_religion]"
+		feedback_set_details("religion_name","[new_religion]")
 
-			var/accepted = 0
-			var/outoftime = 0
-			spawn(200) // 20 seconds to choose
-				outoftime = 1
-			var/new_book_style = "Bible"
+	spawn(1)
```
Comment (skull132) [inline code/game/jobs/job/civilian_chaplain.dm]: Considering the size of the block under spawn(), might be worth refactoring into a separate proc and putting it into a CALLBACK instead. Minor optimizations.

File: code/game/jobs/job/job.dm [hunk 4]
```diff
@@ -29,41 +28,36 @@
...
+//Only override this proc
+/datum/job/proc/after_spawn(mob/living/carbon/human/H)
+
+/datum/job/proc/announce(mob/living/carbon/human/H)
+ 
+/datum/job/proc/equip(mob/living/carbon/human/H, visualsOnly = FALSE, announce = TRUE, alt_title = null)
+	if(!H)
+		return 0
+
+	log_debug("Called equip on [H]")
```
Comment (skull132) [inline code/game/jobs/job/job.dm]: Necessary or nay?

File: code/datums/outfits/outfit.dm (hunk 2 above)
Comment (Arrow768) [reply code/datums/outfits/outfit.dm]: That´s there to ensure that items that might be equipped in post_equip also have the fingerprints applied

**Comment by skull132** [comment]
Following jobs spawned with broken access during live test:
- Medical doctor
- Xenobiologist
- Janitor

File: code/game/jobs/job/civilian.dm [hunk 5]
```diff
@@ -194,20 +198,16 @@
 	selection_color = "#dddddd"
 	access = list(access_janitor, access_maint_tunnels, access_engine, access_research, access_sec_doors, access_medical)
 	minimal_access = list(access_janitor, access_maint_tunnels, access_engine, access_research, access_sec_doors, access_medical)
+	outfit = /datum/outfit/job/mining
```
Comment (skull132) [inline code/game/jobs/job/civilian.dm]:
```suggestion
	outfit = /datum/outfit/job/janitor
```

**Comment by Alberyk** [comment]
The preview sprite at the character setup menu is not displaying custom loadout gear.

**Comment by Sindorman** [comment]
#6046 and #6047 Are also related to this live test merge

File: code/game/jobs/job/civilian_chaplain.dm [hunk 6]
```diff
@@ -12,151 +12,156 @@
...
+							T.set_dir(2)
+
+		H.update_inv_l_hand() // so that it updates the bible's item_state in his hand
+
+		switch(input(H,"Look at your bible - is this what you want?") in list("Yes","No"))
+			if("Yes")
+				accepted = 1
+			if("No")
+				if(outoftime)
+					H << "Welp, out of time, buddy. You're stuck. Next time choose faster."
```
Comment (skull132) [inline code/game/jobs/job/civilian_chaplain.dm]:
```suggestion
					to_chat(H, "Welp, out of time, buddy. You're stuck. Next time choose faster.")
```

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 7]
```diff
@@ -1256,6 +1252,15 @@ var/global/list/obj/item/device/pda/PDAs = list()
 			user << "<span class='notice'>You slide \the [C] into \the [src].</span>"
 	return
 
+/obj/item/device/pda/proc/update_userinfo(var/obj/item/weapon/card/id/idcard, var/mob/living/user)
+	owner = idcard.registered_name
+	ownjob = idcard.assignment
+	ownrank = idcard.rank
+	name = "PDA-[owner] ([ownjob])"
+	if(user)
+		user << "<span class='notice'>Card scanned.</span>"
```
Comment (skull132) [inline code/game/objects/items/devices/PDA/PDA.dm]:
```suggestion
		to_chat(user, "<span class='notice'>Card scanned.</span>")
```

File: code/modules/mob/inventory.dm [hunk 8]
```diff
@@ -43,7 +43,29 @@
 
 //This is just a commonly used configuration for the equip_to_slot_if_possible() proc, used to equip people when the rounds tarts and when events happen and such.
 /mob/proc/equip_to_slot_or_del(obj/item/W as obj, slot)
-	. = equip_to_slot_if_possible(W, slot, TRUE, TRUE, FALSE, ignore_blocked = TRUE)
+	. = equip_to_slot_if_possible(W, slot, TRUE, TRUE, FALSE, TRUE)
+
+// Convinience proc.  Collects crap that fails to equip either onto the mob's back, or drops it.
+// Used in job equipping so shit doesn't pile up at the start loc.
+/mob/living/carbon/human/proc/equip_or_collect(var/obj/item/W, var/slot)
+	if(!W || !istype(W, /obj/item))
```
Comment (skull132) [inline code/modules/mob/inventory.dm]:
```suggestion
	if(!istype(W))
```

`istype(null, /any/path) == false`

**Pull request merged**